llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -688,22 +687,6 @@ END_TWO_BYTE_PACK()
   /// \<target\>ISD namespace).
   bool isTargetOpcode() const { return NodeType >= ISD::BUILTIN_OP_END; }
 
-  /// Test if this node has a target-specific opcode that may raise
-  /// FP exceptions (in the \<target\>ISD namespace and greater than
-  /// FIRST_TARGET_STRICTFP_OPCODE).  Note that all target memory
-  /// opcode are currently automatically considered to possibly raise
-  /// FP exceptions as well.
-  bool isTargetStrictFPOpcode() const {
-    return NodeType >= ISD::FIRST_TARGET_STRICTFP_OPCODE;
```

Comment:
If you're going to start table generating the enum values for this, can you just set a high bit in the enum value for memory / strictfp?

Reply:
Unless I misunderstood you, it would require clearing these bits on any use of opcode, e.g. when creating a node or using the enum value as an index into the [generated] table.

Reply:
No, the enum values would just be larger values. For table compression you could implicitly know the chain status and truncate the table value

Reply:
I see your point now. This may work well for strict-fp opcodes, but it will be more difficult to implement for memory opcodes.

Not all memory opcodes will be generated, because some targets do not provide a tablegen description for all memory nodes. This means that some enum members will have to be assigned a value by hand, and this won't look pretty.

Some target that provide tablegen descriptons for memory nodes do not set SDNPMemOperand property on them. As a result, the generated enum values will not have the "memory bit" set, and there is no way to change the enum value included from a generated file.

In the future, we may want to add more properties (e.g., `isBinOp`), which I think is by itself a good justification for adding a separate field for them rather than encoding them in the opcode.

Reply:
> Not all memory opcodes will be generated, because some targets do not provide a tablegen description for all memory nodes. 

This sounds more like a bug. Your RFC sold not requiring targets to use this as a feature, but I'd consider it a defect. It would be better if targets were required to generate this kind of information from tablegen

Reply:
> > Not all memory opcodes will be generated, because some targets do not provide a tablegen description for all memory nodes.
> 
> This sounds more like a bug.

Depends how you look at it. It wasn't a bug before the proposed changes; targets were not required to describe all target-specific opcodes in *.td files.

> Your RFC sold not requiring targets to use this as a feature, but I'd consider it a defect.

Yes, otherwise it wouldn't be sold. This feature/defect allows gradual migration.

> It would be better if targets were required to generate this kind of information from tablegen

I guess we can do that once all in-tree targets have migrated. Probably need to give some time to downstream targets, too.